sdk/nodejs/servicebus/zMixins.ts

```diff
@@ -306,7 +306,9 @@ export class ServiceBusFunction extends appservice.Function<ServiceBusContext, s
         }
 
         const namespaceName = (args.queue && args.queue.namespaceName) || args.topic!.namespaceName;
+        const namespaceId = (args.queue && args.queue.namespaceName) ? args.queue.id : args.topic!.id;
         const resourceGroupName = (args.queue && args.queue.resourceGroupName) || args.topic!.resourceGroupName;
+        const resourceGroupId = (args.queue && args.queue.resourceGroupName) ? args.queue.id : args.topic!.id;

```

Comment:
1. I think the names are a bit confusing: they are both a queue/topic ID, not namespace ID or resource group ID. 
2. Why do we need both?

Reply:
> I think the names are a bit confusing: they are both a queue/topic ID, not namespace ID or resource group ID.

Yes, but that's not how they're used.

> Why do we need both?

I would love it if we didn't, but it's not clear from the conditions whether or not it is safe to take _either_ the queue ID _or_ the topic ID (i.e. it's not clear whether those properties are mutually exclusive). If it is the case that the user must specify exactly one of those, then we don't need both.

Reply:
Yes, the idea is that exactly one of the (queue, topic) has to be specified, which is enforced in the constructor [here](https://github.com/pulumi/pulumi-azure/blob/master/sdk/nodejs/servicebus/zMixins.ts#L292-L298)

Reply:
Ah, perfect.